src/vm.rs

```diff
@@ -640,9 +753,7 @@ impl<'a, S: Storage, C: PevmChain> Vm<'a, S, C> {
 
                 drop(evm); // release db
 
-                if db.is_lazy {
-                    self.mv_memory.add_lazy_addresses([*from, *to.unwrap()]);
-                }
+                self.mv_memory.add_lazy_locations(lazy_locations);
```

Comment:
Make sure either call site or the implementation checks for an empty array to avoid a meaningless lock.